core/src/org/pentaho/di/core/extension/ExtensionPointMap.java

```diff
@@ -90,7 +90,10 @@ public static ExtensionPointMap getInstance() {
 
   /**
    * Add the extension point plugin to the map
-   * 
+   *
+   * Warning: Do not call this method directly through ExtensionPointMap, it may lead to concurrency problems. Call
+   * PluginRegistry.getInstance().registerPlugin() instead.
+   *
    * @param extensionPointPlugin
    */
   public void addExtensionPoint( PluginInterface extensionPointPlugin ) {
```

Comment:
I don't see how this change addresses the issue. Anyone caling this method will potentially hit the concurrency issue. Access to the map is still not synchronized.